esp-hal/src/lib.rs

```diff
@@ -215,17 +215,34 @@ pub enum Cpu {
 
 /// Which core the application is currently executing on
 pub fn get_core() -> Cpu {
-    Cpu::from_repr(get_raw_core()).unwrap()
+    // This works for both RISCV and Xtensa because both
+    // get_raw_core functions return zero, _or_ something
+    // greater than zero; 1 in the case of RISCV and 0x2000
+    // in the case of Xtensa.
+    match get_raw_core() {
+        0 => Cpu::ProCpu,
+        #[cfg(all(multicore, riscv))]
+        1 => Cpu::AppCpu,
+        #[cfg(all(multicore, xtensa))]
+        0x2000 => Cpu::AppCpu,
+        _ => unreachable!(),
+    }
 }
 
+/// Returns the raw value of the mhartid register.
 #[cfg(riscv)]
 fn get_raw_core() -> usize {
     riscv::register::mhartid::read()
 }
 
+/// Returns the result of reading the PRID register logically ANDed with 0x2000,
+/// the 13th bit in the register. Espressif Xtensa chips use this bit to
+/// determine the core id.
+///
+/// Returns either 0 or 0x2000
```

Comment:
Should say that we should never return xxx because there is an assumption in xx file xx func?